snuba/snapshots/bulk_load.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+import logging
+
+from snuba.clickhouse import ClickhousePool
+from snuba.snapshots import BulkLoadSource
+
+
+class BulkLoader(ABC):
+    """
+    Loads data from a source into a Snuba dataset.
+
+    Implementations can be dataset specific.
+    The dataset returns an instance of this class to be used to perform
+    the bulk load operation.
+    """
+    @abstractmethod
+    def load(self) -> None:
+        raise NotImplementedError
+
+
+class SingleTableBulkLoader(BulkLoader):
+    """
+    Load data from a source table into one clickhouse destination table.
+    """
+
+    def __init__(self,
+        source: BulkLoadSource,
+        dest_table: str,
+        dataset_table: str,
+    ):
+        self.__source = source
+        self.__dest_table = dest_table
+        self.__dataset_table = dataset_table
```

Comment:
This name is confusing — this isn't the table associated with dataset schema, it's the table from the source that is being used to populate the dataset to the destination table (also not the dataset table.)